src/services/userService.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+const { v4: uuidv4 } = require('uuid');
+const bcrypt = require('bcrypt');
+const { User } = require('../models/User.js');
+const { emailService } = require('./emailService.js');
+const { ApiError } = require('../exceptions/ApiError.js');
+
+function validateEmail(email) {
+  const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+
+  if (!email) {
+    throw ApiError.BadRequest('Email is required', {
+      email: 'Email is required',
+    });
+  } else if (!emailRegex.test(email)) {
+    throw ApiError.BadRequest('Email is invalid', {
+      email: 'Email is invalid',
+    });
+  }
+}
+
+function validatePassword(password) {
+  if (!password || password.length < 6) {
+    const errMsg = !password ? 'Password is required' : 'Password is too short';
+
+    throw ApiError.BadRequest(errMsg, { password: errMsg });
+  }
+}
+
+function getByEmail(email) {
+  return User.findOne({ where: { email } });
+}
+
+function getById(userId) {
+  return User.findByPk(userId);
+}
+
+async function register({ name, email, password }) {
+  const existingUser = getByEmail(email);
+
+  if (existingUser) {
+    throw ApiError.BadRequest('User already exists', {
+      email: 'User already exists',
+    });
+  }
+
+  const activationToken = uuidv4();
+  const hashedPassword = await bcrypt.hash(password, 10);
+
+  await User.create({
+    email,
+    password: hashedPassword,
+    name,
+    activationToken,
+  });
+
+  await emailService.sendActivationLink({
+    email,
+    activationToken,
+  });
+}
+
+function generateRestorePasswordToken() {
+  return uuidv4();
+}
+
+function normalize({ id, email }) {
+  return {
+    id,
+    email,
+  };
+}
+
+function updateName(userId, name) {
+  return User.update({ name }, { where: { id: userId } });
+}
+
+function updatePassword(userId, password) {
```

Comment:
Password is being updated without being hashed. Use bcrypt to hash the password before updating it.

Reply:
<img width="528" alt="Screenshot 2023-06-28 at 22 33 55" src="https://github.com/mate-academy/node_auth-app/assets/75370112/90343000-103e-411d-adac-dad8672c77bf">